src/gsi/radarZ_module.f90

```diff
@@ -0,0 +1,2887 @@
+! Force explicit declarations. 
+!-----------------------------------------------------------------------
+
+  IMPLICIT NONE 
+
+!-----------------------------------------------------------------------
+! Variables can be changed by parameter retrieval 
+!-----------------------------------------------------------------------
+
+  fos = 0.5_r_kind      ! Maximum fraction of rain-snow mixture
+  foh = 0.0_r_kind      ! Maximum fraction of rain-hail mixture
+  fog = 0.3_r_kind      ! Maximum fraction of rain-hail mixture
+
+  END SUBROUTINE init_fox_no_hail
+
+  SUBROUTINE model_dsd(n0rain,n0snow,n0hail,n0grpl,rhosnow,rhohail,rhogrpl, &
+             alpharain,alphasnow,alphagrpl,alphahail)
+
+!-----------------------------------------------------------------------
+!
+! PURPOSE:
+!
+! Set dsd values to those used in the arps forecasts
+!
+!-----------------------------------------------------------------------
+!
+! AUTHOR:  Youngsun Jung, 1/20/2006
+!
+!-----------------------------------------------------------------------
+! Force explicit declarations.
+!-----------------------------------------------------------------------
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Define variables:
+!-----------------------------------------------------------------------
+
+
+  REAL(r_kind), INTENT(IN    ) :: n0rain,n0snow,n0hail,n0grpl
+  REAL(r_kind), INTENT(IN    ) :: rhosnow,rhohail,rhogrpl
+  REAL(r_kind), INTENT(IN    ) :: alpharain,alphasnow,alphagrpl,alphahail
+
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+!
+! Beginning of executable code...
+!
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+  N0r=n0rain
+  N0s=n0snow
+  N0h=n0hail
+  N0g=n0grpl
+
+  rhos=rhosnow
+  rhoh=rhohail
+  rhog=rhogrpl
+
+  alphar = alpharain
+  alphas = alphasnow
+  alphag = alphagrpl
+  alphah = alphahail
+
+  END SUBROUTINE model_dsd
+
+  SUBROUTINE coeff_hail(fw)
+
+!-----------------------------------------------------------------------
+!
+! PURPOSE:
+!
+! Scattering matrix coefficient for hail
+!
+!-----------------------------------------------------------------------
+!
+! AUTHOR:  Youngsun Jung, 1/27/2007
+!
+!-----------------------------------------------------------------------
+! Force explicit declarations.
+!-----------------------------------------------------------------------
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Define variables:
+!-----------------------------------------------------------------------
+
+  REAL(r_kind), INTENT(IN   ) :: fw
+  
+  sf = 0.8_r_kind
+
+  sigmah=pi/3*(1-sf*fw)
+  Ah=.125_r_kind*(3+4*exp(-2*sigmah**2)+exp(-8*sigmah**2))
+  Bh=.125_r_kind*(3-4*exp(-2*sigmah**2)+exp(-8*sigmah**2))
+  Ch=.125_r_kind*(1-exp(-8*sigmah**2))
+  Dh=.125_r_kind*(3+exp(-8*sigmah**2))
+  Ckh=exp(-2*sigmah**2)
```

Comment:
Please add _r_kind to 3, 1, 4, 2, and 8
```
pi/3_r_kind*(1_r_kind-sf*fw)
.125_r_kind*(3_r_kind+4_r_kind*exp(-2_r_kind*sigmah**2)+exp(-8_r_kind*sigmah**2))
.125_r_kind*(3_r_kind-4_r_kind*exp(-2_r_kind*sigmah**2)+exp(-8_r_kind*sigmah**2))
.125_r_kind*(1_r_kind-exp(-8_r_kind*sigmah**2))
.125_r_kind*(3_r_kind+exp(-8_r_kind*sigmah**2))
exp(-2_r_kind*sigmah**2)
```

Reply:
We changed the code as you suggested.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-fa0ecb3836c32fcbab1f64a2be08a2292b9c3414d702d475609b32f8205ffcf1R615-R621